src/openrct2-ui/windows/Staff.cpp

```diff
@@ -119,1240 +121,1169 @@ static rct_widget* window_staff_page_widgets[] = {
+        {
+            case WIDX_LOCATE:
+            {
+                if (dropdownIndex == 0)
+                {
+                    ScrollToViewport();
+                }
+                else if (dropdownIndex == 1)
+                {
+                    FollowPeep();
+                }
+                break;
             }
-            else
+            case WIDX_PATROL:
             {
-                auto staffId = EntityId::FromUnderlying(w->number);
-                if (WindowPatrolAreaGetCurrentStaffId() == staffId)
+                // Clear patrol
+                if (dropdownIndex == 1)
                 {
+                    if (_staff == nullptr)
+                    {
+                        return;
+                    }
+
                     window_close_by_class(WC_PATROL_AREA);
+
+                    auto staffSetPatrolAreaAction = StaffSetPatrolAreaAction(
+                        _staff->sprite_index, {}, StaffSetPatrolAreaMode::ClearAll);
+                    GameActions::Execute(&staffSetPatrolAreaAction);
                 }
                 else
                 {
-                    WindowPatrolAreaOpen(staffId);
+                    auto staffId = EntityId::FromUnderlying(number);
+                    if (WindowPatrolAreaGetCurrentStaffId() == staffId)
+                    {
+                        window_close_by_class(WC_PATROL_AREA);
+                    }
+                    else
+                    {
+                        WindowPatrolAreaOpen(staffId);
+                    }
                 }
+                break;
             }
-            break;
         }
     }
-}
-
-static void WindowStaffShowLocateDropdown(rct_window* w, rct_widget* widget)
-{
-    gDropdownItems[0].Format = STR_LOCATE_SUBJECT_TIP;
-    gDropdownItems[1].Format = STR_FOLLOW_SUBJECT_TIP;
-
-    WindowDropdownShowText(
-        { w->windowPos.x + widget->left, w->windowPos.y + widget->top }, widget->height() + 1, w->colours[1], 0, 2);
-    gDropdownDefaultIndex = 0;
-}
-
-static void WindowStaffFollow(rct_window* w)
-{
-    rct_window* w_main = window_get_main();
-    window_follow_sprite(w_main, EntityId::FromUnderlying(w->number));
-}
 
-/**
- * Update the animation frame of the tab icon.
- *  rct2: 0x6BE602
- */
-void WindowStaffOverviewUpdate(rct_window* w)
-{
-    int32_t newAnimationFrame = w->var_496;
-    newAnimationFrame++;
-    if (newAnimationFrame >= 24)
+    void OverviewPrepareDraw()
     {
-        newAnimationFrame = 0;
-    }
-    w->var_496 = newAnimationFrame;
-    widget_invalidate(w, WIDX_TAB_1);
-}
+        if (_staff == nullptr)
+        {
+            return;
+        }
 
-/**
- *
- *  rct2: 0x006BE814
- */
-static void WindowStaffSetOrder(rct_window* w, int32_t order_id)
-{
-    const auto peep = GetStaff(w);
-    if (peep == nullptr)
-    {
-        return;
-    }
+        /**
+         * rct2: 0x006BED21
+         * Disable the _staff pickup if not in pickup state.
+         */
```

Comment:
Remove the doc comment, replace with normal one.

```suggestion
         // Disable the _staff pickup if not in pickup state.
```